presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -176,8 +176,14 @@ protected RelationPlan visitTable(Table node, SqlPlannerContext context)
         Scope scope = analysis.getScope(node);
 
         if (namedQuery != null) {
-            QualifiedObjectName name = createQualifiedObjectName(session, node, node.getName());
-            session.getCteInformationCollector().addCTEReference(node.getName(), analysis.isView(name));
+            if ((session.getCatalog().isPresent() && session.getSchema().isPresent()) ||
```

Comment:
This isn't quite right, and it doesn't support e.g. if you had only the catalog set and a partially qualified name that only specifies the schema.   The real problem here is that unlike views, regular with query CTEs don't have to have fully qualified names.

I would instead add an additional utility method called something like `hasFullyQualifiedName()`, that basically checks whether a qualified name can be created (catalog and schema are specified either in the node name or in the session).

then i would say
```
// with query CTEs don't need to be able to be fully qualified, as they aren't a part of any catalog or schema
boolean isView = hasFullyQualifiedName(session, node.getName) ? analysis.isView(createQualifiedObjectName(session, node, node.getName()) : false;
session.getCteInformationCollector().addCTEReference(node.getName(), isView);
```

Reply:
Ugh, I'd been testing with the CLI and missed this edge case. Cool, will do the refactoring as suggested and add tests

Reply:
Actually, there's a case where `hasFullyQualifiedName` will fail as well. Consider the following -
```

$ presto-cli --server localhost:8086 --catalog 'local_hms' --schema 'tpch_sf1_parquet'

presto:tpch_sf1_parquet> set session verbose_optimizer_info_enabled=true;
SET SESSION
presto:tpch_sf1_parquet> create or replace view local_hms.tpch_sf1_parquet.t1 as select orderkey from tpch.sf1.orders;
CREATE VIEW
presto:tpch_sf1_parquet> explain with t1 as (select 42 as t1_cte_col) select * from t1, local_hms.tpch_sf1_parquet.t1;
                                                                                                                                                                                                                                                                                        >
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
 - Output[PlanNodeId 17][t1_cte_col, orderkey] => [expr:integer, orderkey:bigint]                                                                                                                                                                                                       >
         Estimates: {source: CostBasedSourceInfo, rows: 1500000 (20.03MB), cpu: 48000010.00, memory: 5.00, network: 21000005.00}                                                                                                                                                        >
         t1_cte_col := expr (1:46)                                                                                                                                                                                                                                                      >
     - RemoteStreamingExchange[PlanNodeId 323][GATHER] => [orderkey:bigint, expr:integer]                                                                                                                                                                                               >
             Estimates: {source: CostBasedSourceInfo, rows: 1500000 (20.03MB), cpu: 48000010.00, memory: 5.00, network: 21000005.00}                                                                                                                                                    >
         - CrossJoin[PlanNodeId 12] => [orderkey:bigint, expr:integer]                                                                                                                                                                                                                  >
                 Estimates: {source: CostBasedSourceInfo, rows: 1500000 (20.03MB), cpu: 48000010.00, memory: 5.00, network: 5.00}                                                                                                                                                       >
                 Distribution: REPLICATED                                                                                                                                                                                                                                               >
             - TableScan[PlanNodeId 6][TableHandle {connectorId='tpch', connectorHandle='orders:sf1.0', layout='Optional[orders:sf1.0]'}] => [orderkey:bigint]                                                                                                                          >
                     Estimates: {source: CostBasedSourceInfo, rows: 1500000 (20.03MB), cpu: 13500000.00, memory: 0.00, network: 0.00}                                                                                                                                                   >
                     orderkey := tpch:orderkey (3:3)                                                                                                                                                                                                                                    >
                     tpch:orderstatus                                                                                                                                                                                                                                                   >
                         :: [["F"], ["O"], ["P"]]                                                                                                                                                                                                                                       >
             - LocalExchange[PlanNodeId 355][SINGLE] () => [expr:integer]                                                                                                                                                                                                               >
                     Estimates: {source: CostBasedSourceInfo, rows: 1 (14B), cpu: 5.00, memory: 0.00, network: 5.00}                                                                                                                                                                    >
                 - RemoteStreamingExchange[PlanNodeId 322][REPLICATE] => [expr:integer]                                                                                                                                                                                                 >
                         Estimates: {source: CostBasedSourceInfo, rows: 1 (14B), cpu: 5.00, memory: 0.00, network: 5.00}                                                                                                                                                                >
                     - Project[PlanNodeId 1][projectLocality = LOCAL] => [expr:integer]                                                                                                                                                                                                 >
                             Estimates: {source: CostBasedSourceInfo, rows: 1 (14B), cpu: 5.00, memory: 0.00, network: 0.00}                                                                                                                                                            >
                             expr := INTEGER'42'                                                                                                                                                                                                                                        >
                         - LocalExchange[PlanNodeId 354][ROUND_ROBIN] () => []                                                                                                                                                                                                          >
                                 Estimates: {source: CostBasedSourceInfo, rows: 1 (14B), cpu: 0.00, memory: 0.00, network: 0.00}                                                                                                                                                        >
                             - Values[PlanNodeId 0] => []                                                                                                                                                                                                                               >
                                     Estimates: {source: CostBasedSourceInfo, rows: 1 (14B), cpu: 0.00, memory: 0.00, network: 0.00}                                                                                                                                                    >
                                     ()                                                                                                                                                                                                                                                 >
 Triggered optimizers: [PruneTableScanColumns, UnaliasSymbolReferences, RemoveRedundantIdentityProjections, PruneUnreferencedOutputs, PickTableLayoutWithoutPredicate, PruneUnreferencedOutputs, UnaliasSymbolReferences, PruneUnreferencedOutputs, PruneUnreferencedOutputs, PruneUnref>
 Applicable optimizers: [RandomizeNullKeyInOuterJoin, MergePartialAggregationsWithFilter]                                                                                                                                                                                               >
 CTEInfo: [local_hms.tpch_sf1_parquet.t1: 1 (is_view: true), t1: 1 (is_view: true)]   
```

Since we have the catalog and session specified on the Session, and we are including the the view `local_hms.tpch_sf1_parquet.t1` in the query, the  `t1` CTE is incorrectly also tagged as  view and we get **CTEInfo: [local_hms.tpch_sf1_parquet.t1: 1 (is_view: true), t1: 1 (is_view: true)]** as the result

This is because of the way we [keep track of active views](https://github.com/prestodb/presto/blob/3aa85ea3cfa8c1a9e717e786575ab491b07668e1/presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java#L748-L756) in the query. @mlyublena Do you have any ideas on how to fix this ?

Reply:
I think we are better off reverting https://github.com/prestodb/presto/pull/20622 for now while we work on a better fix..

Reply:
That's a great catch. i think the complete solution will have to use the "Scope" field somehow or perhaps moving all the logic to the statement analyzer where we are already checking if something is a withQuery/view/materialized view, so can just log there.

I think it's okay to revert for now (but make sure to put full information of why it's being reverted and what the bugs are in the commit message and PR description for future reference)

Reply:
@aaneja good catch!
please revert the original fix, I'll try to find a proper solution later!

Reply:
Reverting done here: https://github.com/prestodb/presto/pull/20744